install_config/configuring_authentication.adoc

```diff
@@ -1391,3 +1391,55 @@ a secret with 32 or 64 bytes.
 <3> Encrypting secret, used to encrypt sessions. Must be 16, 24, or 32
 characters long, to select AES-128, AES-192, or AES-256.
 ====
+
+[[configuring-authentication-security]]
+== Security
+
+Administrators can prevent clients from accessing the API by their `User-Agent`
+header in the new `*userAgentMatching*` configuration setting.
+
+For example, you can deny clients using particular libraries or binaries. This
+example denies the Kubernetes 1.2 client binary, OpenShift Origin 1.1.3
+binary, and the POST and PUT *httpVerbs*:
+
+====
+----
+policyConfig:
+  userAgentMatchingConfig:
+    defaultRejectionMessage: "Your client is too old.  Go to https://example.org to update it."
+    deniedClients:
+    - regex: '\w+/v(?:(?:1\.1\.1)|(?:1\.0\.1)) \(.+/.+\) openshift/\w{7}'
+    - regex: '\w+/v(?:1\.1\.3) \(.+/.+\) openshift/\w{7}'
+    - regex: '\w+/v1\.2\.0 \(.+/.+\) kubernetes/\w{7}'
+      httpVerbs:
+      - POST
+      - PUT
```

Comment:
>  I think I addressed your comments, but not sure I fully understand your reference to line 1412

Sorry for not stating it clearly. Let's see an example:
```
    deniedClients:
    - regex: '<1st regular expression>'
    - regex: '\w+/v(?:1\.1\.3) \(.+/.+\) openshift/\w{7}'
      httpVerbs:
      - <verb1>
      - <verb2>
    - regex: '\w+/v1\.2\.0 \(.+/.+\) kubernetes/\w{7}'
      httpVerbs:
      - <verb3>
      - <verb4>
```
The 1st `regex` does not specify `httpVerbs` (by default, for a `regex`, not specifying httpVerbs == specifying all verbs).
The 2nd `regex` specifies verb1 and verb2.
The 3rd `regex` specifies verb3 and verb4.
Such configuration will:
deny requests of all verbs from client who (`User-Agent` header) matches 1st `regex`,
deny requests of verb1 and verb2 from client who matches the 2nd `regex`,
deny requests of verb3 and verb4 from client who matches the 3rd `regex`.

As per such explanation, your configuration will:
deny openshift 1.1.1 and 1.0.1 client binaries' all http verbs (GET, DELETE ....),
deny openshift 1.1.3 client binary's all http verbs (GET, DELETE ....),
deny kubernetes 1.2.0 client binary's POST and PUT httpVerbs

But the wording:
> +example denies the Kubernetes 1.2 client binary, OpenShift Origin 1.1.3
> +binary, and the POST and PUT *httpVerbs*:
means: for openshift 1.1.3 client binary, only POST and PUT are denied (i.e. other verbs GET, DELETE are not denied).
So my suggestion was:
```
+    - regex: '\w+/v(?:1\.1\.3) \(.+/.+\) openshift/\w{7}'
+      **- POST**
+      **- PUT**
+    - regex: '\w+/v1\.2\.0 \(.+/.+\) kubernetes/\w{7}'
+      httpVerbs:
+      - POST
+      - PUT
```